test/unit/timeout.test.ts

```diff
@@ -0,0 +1,89 @@
+import { setTimeout } from 'node:timers/promises';
+
+import { expect } from 'chai';
+
+import { CSOTError, MongoInvalidArgumentError, Timeout } from '../mongodb';
+
+describe('Timeout', function () {
+  let timeout: Timeout;
+
+  beforeEach(() => {
+    if (Timeout.is(timeout)) {
+      timeout.clear();
+    }
+  });
+
+  describe('constructor()', function () {
+    context('when called with a duration of 0', function () {
+      it('does not create a timeout instance (creates infinite timeout)', function () {
+        timeout = Timeout.expires(0);
+        expect(timeout).to.not.have.property('id');
+      });
+    });
+
+    context('when called with a duration greater than 0', function () {
+      beforeEach(() => {
+        timeout = Timeout.expires(2000);
+      });
+      it('creates a timeout instance that will not keep the Node.js event loop active', function () {
+        expect(timeout).to.have.property('id');
+        const id = timeout['id'];
+        expect(id?.hasRef()).to.be.false;
```

Comment:
```suggestion
        // @ts-expect-error: accessing private property
        const id = timeout.id;
```

Rather than avoid the TS error, I'd rather acknoledge it and clean up the code if it were no longer disallowed.